src/main/java/com/launchdarkly/client/StreamProcessor.java

```diff
@@ -34,6 +42,8 @@
     this.config = config;
     this.sdkKey = sdkKey;
     this.requestor = requestor;
+    this.heartbeatDetectorService = Executors.newScheduledThreadPool(1);
```

Comment:
Elsewhere we used named threadpools like this:
```    
ThreadFactory threadFactory = new ThreadFactoryBuilder()
        .setNameFormat("LaunchDarkly-HeartbeatDetector-%d")
        .build();
    this. heartbeatDetectorService = Executors.newSingleThreadScheduledExecutor(threadFactory);
```
This makes for easier debugging and reading of log statements.
Also, I think we want this to be a single threaded executor, even though practically it won't matter much.